cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -0,0 +1,1146 @@
+//! Implementation of a standard AArch64/x64-like ABI.
```

Comment:
The general comment structure of "this is for AArch64 and X64" gives me some concern.  These comments are soon going to be out of date and hence confusing, once ports to other targets exist.  I would prefer that

* the implementation is introduced as "a vanilla modern ABI, in which some the first few (possibly zero) args are passed in registers, and the rest on the stack".  I think that will cover most bases in the short/medium term, at least until we have to deal with the POWER64 ABIs, which are pretty weird due to the TOC requirement.

* references to "AArch64 and X64" are removed

* in the comment(s), instantiations/descriptions for each architecture are laid out explicitly, so that when adding a new architecture, the relevant pieces of the comments can be updated